### ISSUE TITLE
Tabs ViewModel are recreated but not used

### ISSUE THREAD

**Issue by DavidMarquezF** [MvvmCross/MvvmCross#3682]
## 🐛 Bug Report

When using tabs the same way as the playground project, the ViewModels are recreated but not used every time you navigate to them.

I found out that the ViewModels get created and they are correctly added to the `MvxCachingFragmentStatePagerAdapter.` The problem seems to be in `MvxCachingFragmentPagerAdapter,` because the `InstantiateItem` function uses the fragments found in the fragmentManager and doesn't get to the point where it calls `GetItem` (where it would instantiate the view with the new ViewModel).

https://github.com/MvvmCross/MvvmCross/blob/4684ddba5b605c1bc2c4322251c8fb5ff41cf819/MvvmCross.Android.Support/Fragment/MvxCachingFragmentPagerAdapter.cs#L88-L97 

As you can see the commented lines say that it should effectively reuse the fragments created before, but if it should, it shouldn't recreate the ViewModels.
### Expected behavior

Either the ViewModel should be recreated and used or it should just keep using the last ViewModel without creating the old one. I don't know which would be the best behavior (in my case, for example, it would be to recreate it each time you call navigate).

### Reproduction steps

1. Add a variable that changes each time Tab1ViewModel is created (like a static counter)
2. Bind it to the view (this way you can see what ViewModel instance is the view currently using)
3. Place a breakpoint in the constructor of the ViewModel (or the Initialize method)
4. Navigate to Master/Detail and press Show Tabs
5. Navigate to Master/Detail Child and then close it
6. Initialize method is called but the Tab1View keeps showing the last View Model value

### Configuration

**Version:** 6.4.1

**Platform:** 
- [ ] :iphone: iOS
- [x] :robot: Android
- [ ] :checkered_flag: WPF
- [ ] :earth_americas: UWP
- [ ] :apple: MacOS
- [ ] :tv: tvOS
- [ ] :monkey: Xamarin.Forms

**Comment by DavidMarquezF** [comment]
In general tab navigation seems a little strange overall because you need to navigate to the tabs to show them, but calling Navigate implies creating a new ViewModel. Moreover, you have to call ShowInitialViewModel command from the view which feels a little hacky. Shouldn't the ViewModel be in charge of all the navigation? This also means that when the orientation is changed or you go back to the tabs screen the ShowInitialCommand is called again (because it is handled in the view) and it creates new ViewModels.

I would really like to get this fixed since it affects some of our projects and I believe that this is something that a lot of apps will find. But before trying to do anything I thought that it would be best to get some feedback and discuss this a little bit (maybe I'm missing or I don't understand something) 😄

**Comment by Cheesebaron** [comment]
If you want to navigate between tabs, you should use `ChangePresentation` with a hint to which tab you are navigating to.

```
NavigationService.ChangePresentation(new MvxPagePresentationHint(typeof(Tab2ViewModel))));
```

**Comment by DavidMarquezF** [comment]
Maybe I haven't explained myself correctly. The problem is when rotating the screen or coming back from another screen, not changing tabs once they are already shown.

**Comment by Cheesebaron** [comment]
That implies the fragment was killed off while you were showing something else. Seems like normal behavior to me.

**Comment by DavidMarquezF** [comment]
Yes but the ViewModel shouldn't get recreated. Since the tab root is not destroyed, it keeps the tab fragment in its ChildFragmentManager so the tabs ViewModels shouldn't be created. 

Now, what happens is that a new View Model for each tab is created and Initialized (which in my case, loads a lot of info) but the old View Model is reused. 

This can be reproduced in the playground project following the reproduction steps.
